### DISCUSSION TITLE
Provide values to ActionInputError

### DISCUSSION THREAD

**Discussion by vixalien** [withastro/roadmap#1071] [Proposal]
### Body

# Summary

`ActionInputError` should provide an additional `values` field and constructor property

# Background & Motivation

`ActionInputError` is a class that's meant to be thrown when astro encounters an error while validating the input. However, since Astro doesn't provide any other validation beside from zod, which is limiting in it's own regard (you can't validate for example unique fields in a database using this way), we resolved to creating an `ActionInputError` to demonstrate these other errors.

However, when you use that error, and you catch it in an astro component, any form data will be cleared, which can be detrimental for long forms which the user will now have to fill again for a database error.

It would be beneficial to provide `.values` on `ActionInputError` to help in these situations. 

# Goals

- Help users not have to fill in forms again when a database input validation fails (for example, unique fields)

# Example

```js
const error = new ActionInputError(issues, values);
```

**Comment by rvanlaarhoven** [comment]
Instead of manually having to pass the inputs back to an `ActionInputError`, another option would be to return the inputs as part of the `ActionResult`. Ideally the inputs are returned not just on input errors, but in any action error result.

This would also be helpful when using React's `useActionState` action passed to a form's `action` prop, which also clears the form on a submit.
